test/cql-pytest/test_uda.py

```diff
@@ -86,4 +92,32 @@ def test_incorrect_state_func(scylla_only, cql, test_keyspace):
     with new_function(cql, test_keyspace, avg2_partial_body) as avg2_partial, new_function(cql, test_keyspace, div_body) as div_fun:
         custom_avg_body = f"(bigint) SFUNC {avg2_partial} STYPE tuple<bigint, bigint> FINALFUNC {div_fun} INITCOND (0,0)"
         with pytest.raises(InvalidRequest, match="State function not found"):
-            cql.execute(f"CREATE AGGREGATE {test_keyspace}.{unique_name()} {custom_avg_body}")
\ No newline at end of file
+            cql.execute(f"CREATE AGGREGATE {test_keyspace}.{unique_name()} {custom_avg_body}")
+
+# Test that UDA works without final function and returns accumulator then
+def test_no_final_func(scylla_only, cql, test_keyspace):
+    schema = "id int primary key"
+    sum_partial_body = "(acc int, val int) RETURNS NULL ON NULL INPUT RETURNS int LANGUAGE lua AS 'return acc + val'"
+    with new_test_table(cql, test_keyspace, schema) as table:
+        for i in range(5):
+            cql.execute(f"INSERT INTO {table} (id) VALUES ({i})")
+        with new_function(cql, test_keyspace, sum_partial_body) as sum_partial:
+            custom_sum_body = f"(int) SFUNC {sum_partial} STYPE int INITCOND 0"
+            with new_aggregate(cql, test_keyspace, custom_sum_body) as custom_sum:
+                custom_res = cql.execute(f"SELECT {custom_sum}(id) AS result FROM {table}")[0]
+                avg_res = cql.execute(f"SELECT sum(id) AS result FROM {table}")[0]
+                assert custom_res == avg_res
+
+# Test that UDA works without initcond and inits acc as null
+def test_no_initcond(scylla_only, cql, test_keyspace):
+    schema = "id int primary key"
+    state_body = "(acc int, val int) CALLED ON NULL INPUT RETURNS int LANGUAGE lua AS 'if acc == null then return null else return acc + val end'"
```

Comment:
Let's make this function return something else than null when `acc == null`, e.g. 42, otherwise it will always return the same value if initcond is null, regardless of how many rows it goes through. And then I'm worried we could easily miss a regression - e.g. that the aggregate is not called at all if initcond == null.

Reply:
42 might not be the best choice, but `val` would be quite a good candidate, as in:
```lua
if acc == null then return val else return acc + val end
```